### ISSUE TITLE
Account linking works for one database but not the other, same settings.

### ISSUE THREAD

**Issue by xsorifc28** [firebase/FirebaseUI-Android#989]
I have a debug and release firebase project for one of my apps. I use login with google/fb/email. In one of the firebase projects, when I sign in with google + fb, i see that auth joins the accounts (debug). However, in the other database (release), this does not happen. Both projects have "One account per email address" enabled. The release firebase project was imported as a  legacy firebase project, this is the only reason I though that the behavior could be different.

**Comment by samtstern** [comment]
@xsorifc28 we will need more information to debug:

  * Have you enabled the correct Sign In Providers in each project?
  * Do you have the SHA1 configured properly?  This is likely related to the issue since Google Sign In is dependent on the SHA1
  * What error logs do you see?

**Comment by xsorifc28** [comment]
@samtstern 

- Both projects have the same Sign In Providers enabled.
- One project is for the release version of the app, so that one has the release SHA1 where as the other has the debug SHA1. Sign in with Google/Facebook/Email works on _both_ projects. The **release** project **does not** join the emails. I see two auth-list items for the same email, one for Google and one for Facebook.
- I didn't see any logs that related to the sign-in process.

Release database (again, this database was originally imported from firebase.com, some UIDs are in `provider:number` format.

![2017-10-23 22_13_41-culturecloud authentication firebase console](https://user-images.githubusercontent.com/5721836/31921862-96f6dbc4-b83f-11e7-8b67-34de77025ba0.png)


Debug database:
![2017-10-23 22_13_59-culturecloudbeta authentication firebase console](https://user-images.githubusercontent.com/5721836/31921889-c20439ba-b83f-11e7-8482-51446faa35e3.png)


The emails are the same, just not sure why they are not joined in the release project.

**Comment by samtstern** [comment]
@xsorifc28 ok so is this what you are trying to do:

- Sign up with email
- Sign out
- Sign up with a Google/Facebook account with an email that matches
- See account linking

?

**Comment by xsorifc28** [comment]
Yes.

**Comment by samtstern** [comment]
@xsorifc28 do you see the same UI flow in each app?  You should see one of the "welcome back" screens and then be prompted to sign in with the original method.

In general any information you could provide here would be helpful.  Logging output, a description of where the UI fails, etc.

It's really not clear that this is a FirebaseUI issue at all based on the amount of information provided.

**Comment by xsorifc28** [comment]
Yes, UI flow is same - infact the app is the same, just different package name. The logs just show "notifying about auth state listener". No information about linking.  I also am beginning to believe that this is not a Firebase UI issue. Though, could you point me to where the linking happens in FirebaseUI? I'd like to debug that part of the code to see if I can get any information out of it.

**Comment by demolaf** [comment]
Hello,

Thank you for taking the time to submit this bug.

We are happy to let you know that this has been resolved in **FirebaseUI for Android v10.0.0**, which is now available as a beta release. You can find the release notes [here](https://github.com/firebase/FirebaseUI-Android/releases/tag/10.0.0-beta02) and follow the [upgrade guide](https://github.com/firebase/FirebaseUI-Android/blob/master/docs/upgrade-to-10.0.md) to get started.

We will close this issue now. If you still experience this in the new version, please feel free to open a new issue.

Thank you again for your contribution.
